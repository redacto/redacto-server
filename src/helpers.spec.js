/* eslint-env node, mocha */

// ============================================================
// Import packages
import { assert, expect } from 'chai';
import faker from 'faker';
import proxyquire from 'proxyquire';
import sinon from 'sinon';

// ============================================================
// Import modules
import {
    getHelloMessage,
} from './helpers';

// ============================================================
// Tests
describe('getHelloMessage', () => {
    it('Should return a string', () => {
        const helloMessage = getHelloMessage();

        assert.isString(helloMessage);

        expect(helloMessage).to.matchSnapshot();
    });

    it('Should include name', () => {
        const name = faker.random.alphaNumeric(10);
        const helloMessage = getHelloMessage(name);

        assert.isString(helloMessage);
        assert.isTrue(helloMessage.includes(name));
    });
});

describe('sayHello', () => {
    it('should log the default message if no name provided', () => {
        const consoleLogSpy = sinon.spy();
        const proxy = proxyquire('./helpers', {
            'better-console': {
                log: consoleLogSpy,
            },
        });

        proxy.sayHello();

        assert.isTrue(consoleLogSpy.calledOnce);
        assert.isString(consoleLogSpy.args[0][0]);
    });
});
