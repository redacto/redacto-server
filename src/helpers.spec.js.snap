// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Should return a string 1`] = `"Hello you !"`;

exports[`getHelloMessage Should return a string 1`] = `"Hello you !"`;
